RecoVertex/PrimaryVertexProducer_Alpaka/plugins/alpaka/FitterAlgo.dev.cc

```diff
@@ -0,0 +1,323 @@
+          erry = 1 / s_wy;
+#ifdef DEBUG_RECOVERTEX_PRIMARYVERTEXPRODUCER_ALPAKA_FITTERALGO
+          printf("[FitterAlgo::fitVertices()] After dividing %i iteration %1.9f %1.9f %1.9f %1.9f %1.9f \n",
+                 niter,
+                 x,
+                 y,
+                 z,
+                 errx,
+                 errz);
+          printf("[FitterAlgo::fitVertices()] Compare old and new: %1.9f %1.9f, %1.9f %1.9f, %1.9f %1.9f \n",
+                 old_x,
+                 x,
+                 old_y,
+                 y,
+                 old_z,
+                 z);
+#endif
+          if ((abs(old_x - x) < precision) && (abs(old_y - y) < precision) && (abs(old_z - z) < precision))
+            break;  // If good enough, stop the iterations
+        }  // end while
+        // Assign everything back in global memory to get the fitted vertex!
+        errx *= corr_x * corr_x;
+        erry *= corr_x * corr_x;
+        errz *= corr_z * corr_z;
+        vertices[i].x() = x;
+        vertices[i].y() = y;
+        vertices[i].z() = z;
+        vertices[i].errx() = errx;
+        vertices[i].erry() = erry;
+        vertices[i].errz() = errz;
+        vertices[i].ndof() = ndof;
+        // Last get the chi square of the final vertex fit
+        double chi2 = 0.;
+        for (int itrackInVertex = 0; itrackInVertex < vertices[i].ntracks(); itrackInVertex++) {
+          int itrack = vertices[i].track_id()[itrackInVertex];
+          // Position (ref point) of the track
+          float tx = tracks[itrack].x();
+          float ty = tracks[itrack].y();
+          float tz = tracks[itrack].z();
+          float wx = tracks[itrack].dxy2();
+          float wz = tracks[itrack].dz2();
+          chi2 +=
+              (tx - x) * (tx - x) / (errx + wx) + (ty - y) * (ty - y) / (erry + wx) +
+              (tz - z) * (tz - z) /
+                  (errz +
+                   wz);  // chi2 doesn't use the PCA distance, but the ref point coordinates as in https://github.com/cms-sw/cmssw/blob/master/RecoVertex/PrimaryVertexProducer/interface/WeightedMeanFitter.h#L316
+        }  // end for
+        vertices[i].chi2() = chi2;
+#ifdef DEBUG_RECOVERTEX_PRIMARYVERTEXPRODUCER_ALPAKA_FITTERALGO
+        printf(
+            "[FitterAlgo::fitVertices()] Vertex %i, x: %1.9f, y:%1.9f, z:%1.9f, errx:%1.9f, errz:%1.9f, chi2:%1.9f, "
+            "ndof:%1.9f\n",
+            i,
+            vertices[i].x(),
+            vertices[i].y(),
+            vertices[i].z(),
+            vertices[i].errx(),
+            vertices[i].errz(),
+            vertices[i].chi2(),
+            vertices[i].ndof());
+#endif
+      }  // end for (stride) loop
+    }  // operator()
+  };  // class fitVertices
+
+  FitterAlgo::FitterAlgo(Queue& queue, const int32_t nV, fitterParameters fPar)
+      : useBeamSpotConstraint(cms::alpakatools::make_device_buffer<bool>(queue)) {
+    // Set fitter parameters
+    alpaka::memset(queue, useBeamSpotConstraint, fPar.useBeamSpotConstraint);
```

Comment:
I think a single `bool` to be more economical to be passed by value to the kernel than via a buffer.